internal/serialization/portable_field.go

```diff
@@ -0,0 +1,380 @@
+package serialization
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hazelcast/hazelcast-go-client/serialization"
+	"github.com/hazelcast/hazelcast-go-client/types"
+)
+
+type PortableField struct {
+	Name  string
+	Type  PortableFieldType
+	Value any
+}
+
+func (f PortableField) String() string {
+	return fmt.Sprintf("%s:%s", f.Name, f.formatValue())
+}
+
+func (f PortableField) formatValue() (v string) {
+	defer func() {
+		if e := recover(); e != nil {
+			v = ValueNotDecoded
+		}
+	}()
+	if f.Value == nil {
+		return ""
+	}
+	switch f.Type {
+	case PortableTypeNone:
+	case PortableTypePortable:
+		// nested portable is not supported yet
+		return ValueNotDecoded
+	case PortableTypeByte, PortableTypeBool, PortableTypeUint16,
+		PortableTypeInt16, PortableTypeInt32, PortableTypeInt64,
+		PortableTypeFloat32, PortableTypeFloat64, PortableTypeString,
+		PortableTypePortableArray, PortableTypeByteArray, PortableTypeBoolArray,
+		PortableTypeUInt16Array, PortableTypeInt16Array, PortableTypeInt32Array,
+		PortableTypeInt64Array, PortableTypeFloat32Array, PortableTypeFloat64Array,
+		PortableTypeStringArray:
+		return fmt.Sprintf("%v", f.Value)
+	case PortableTypeDecimal:
+		return MarshalDecimal(f.Value)
+	case PortableTypeDecimalArray:
+		return ValueNotDecoded
+	case PortableTypeTime:
+		sr, err := MarshalLocalTime(f.Value)
+		if err != nil {
+			return ValueNotDecoded
+		} else if sr == nil {
+			return ValueNil
+		} else {
+			return *sr
+		}
+	case PortableTypeTimeArray:
+		return ValueNotDecoded
+	case PortableTypeDate:
+		sr, err := MarshalLocalDate(f.Value)
+		if err != nil {
+			return ValueNotDecoded
+		} else if sr == nil {
+			return ValueNil
+		} else {
+			return *sr
+		}
+	case PortableTypeDateArray:
+		return ValueNotDecoded
+	case PortableTypeTimestamp:
+		sr, err := MarshalLocalDateTime(f.Value)
+		if err != nil {
+			return ValueNotDecoded
+		} else if sr == nil {
+			return ValueNil
+		} else {
+			return *sr
+		}
+	case PortableTypeTimestampArray:
+		return ValueNotDecoded
+	case PortableTypeTimestampWithTimezone:
+		sr, err := MarshalOffsetDateTime(f.Value)
+		if err != nil {
+			return ValueNotDecoded
+		} else if sr == nil {
+			return ValueNil
+		} else {
+			return *sr
+		}
+	case PortableTypeTimestampWithTimezoneArray:
+		return ValueNotDecoded
+	default:
+		return ValueUnknown
+	}
+	return v
+}
+
+// PortableFieldType corresponds to FieldDefinitionType+1
+type PortableFieldType int32
```

Comment:
Why we are not using FieldDefinitionType directly from Go Client?

Reply:
`FieldlDefinitionType` starts from `0` and it means `TypePortable`. The zero value for `int32` is also `0`, so when used in `PortableField` and not setting the `Type` field, it would mean that portable field contains portable data.